### PR TITLE
fix(test): fix nightly access key nonce tests

### LINKED ISSUE #4158: test_cases_testnet_rpc tests fail with nightly features on NayDuck
See:
http://nayduck.eastus.cloudapp.azure.com:3000/#/test/123689
http://nayduck.eastus.cloudapp.azure.com:3000/#/test/123703
http://nayduck.eastus.cloudapp.azure.com:3000/#/test/123707

We missed it due to rust tests were not run with features enabled

### PR DESCRIPTION
**Pull request by bowenwang1996**
Fixes #4158.

Test plan
---------
http://nayduck.eastus.cloudapp.azure.com:3000/#/

### CODE REVIEW

File: test-utils/testlib/src/standard_test_cases.rs [hunk 1]
```diff
@@ -840,6 +840,30 @@ pub fn test_delete_key_last(node: impl Node) {
...
+            key
+        }
+        #[cfg(not(feature = "protocol_feature_access_key_nonce_range"))]
+        {
+            let _ = (user, result);
+            access_key.clone()
+        }
+    };
+    assert_eq!(access_key_view, key.into());
+}
```
Comment (EgorKulikov) [inline test-utils/testlib/src/standard_test_cases.rs:866]:
```suggestion
}

```

File: test-utils/testlib/src/standard_test_cases.rs [hunk 2]
```diff
@@ -840,6 +840,30 @@ pub fn test_delete_key_last(node: impl Node) {
...
+            let mut key = access_key.clone();
+            let block = user.get_block_by_hash(result.transaction_outcome.block_hash);
+            if let Some(b) = block {
+                key.nonce = (b.header.height - 1) * AccessKey::ACCESS_KEY_NONCE_RANGE_MULTIPLIER;
+            }
+            key
+        }
+        #[cfg(not(feature = "protocol_feature_access_key_nonce_range"))]
+        {
+            let _ = (user, result);
```
Comment (EgorKulikov) [inline test-utils/testlib/src/standard_test_cases.rs:861]: I am not sure what is happening here, are we trying to avoid user and result being not used?

**Review by EgorKulikov — APPROVED** [review]

File: test-utils/testlib/src/standard_test_cases.rs [hunk 3]
```diff
@@ -840,6 +840,30 @@ pub fn test_delete_key_last(node: impl Node) {
...
+            access_key.clone()
+        }
```
Comment (matklad) [inline test-utils/testlib/src/standard_test_cases.rs:863]:
```suggestion
    let key = if cfg!(feature = "protocol_feature_access_key_nonce_range") {
            let mut key = access_key.clone();
            let block = user.get_block_by_hash(result.transaction_outcome.block_hash);
            if let Some(b) = block {
                key.nonce = (b.header.height - 1) * AccessKey::ACCESS_KEY_NONCE_RANGE_MULTIPLIER;
            }
            key
        } else  {
            let _ = (user, result);
            access_key.clone()
        }
```

Seems like this *might* be enough? The benefit of `if cfg!()` is that both branches are type-checked (but then one is eliminated during codegen).

File: test-utils/testlib/src/standard_test_cases.rs [hunk 4]
```diff
@@ -840,6 +840,30 @@ pub fn test_delete_key_last(node: impl Node) {
     assert!(node_user.get_access_key(&account_id, &node.signer().public_key()).is_err());
 }
 
+fn assert_access_key(
+    access_key: &AccessKey,
+    access_key_view: AccessKeyView,
+    result: &FinalExecutionOutcomeView,
+    user: Box<dyn User>,
```
Comment (matklad) [inline test-utils/testlib/src/standard_test_cases.rs:847]:
```suggestion
    user: &dyn User,
```

seems like just ref is enough?

File: test-utils/testlib/src/standard_test_cases.rs (hunk 2 above)
Comment (bowenwang1996) [reply test-utils/testlib/src/standard_test_cases.rs:861]: Yes

File: test-utils/testlib/src/standard_test_cases.rs [hunk 5]
```diff
@@ -847,19 +847,19 @@ fn assert_access_key(
...
-            let _ = (user, result);
-            access_key.clone()
+        let mut key = access_key.clone();
+        let block = user.get_block_by_hash(result.transaction_outcome.block_hash);
+        if let Some(b) = block {
+            key.nonce = (b.header.height - 1) * AccessKey::ACCESS_KEY_NONCE_RANGE_MULTIPLIER;
         }
+        key
+    } else {
+        let _ = (user, result);
```
Comment (matklad) [inline test-utils/testlib/src/standard_test_cases.rs:857]:
```suggestion
        let _ = (user, result);
```

You no longer need this

**Pull request merged**